src/main/java/com/jfrog/xray/client/impl/services/graph/GraphImpl.java

```diff
@@ -0,0 +1,95 @@
+import com.fasterxml.jackson.databind.ser.FilterProvider;
+import com.fasterxml.jackson.databind.ser.PropertyWriter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+import com.jfrog.xray.client.impl.XrayClient;
+import com.jfrog.xray.client.impl.util.ObjectMapperHelper;
+import com.jfrog.xray.client.services.graph.Graph;
+import com.jfrog.xray.client.services.graph.GraphResponse;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.util.EntityUtils;
+import org.jfrog.build.extractor.scan.DependencyTree;
+
+import java.io.IOException;
+
+/**
+ * Created by romang on 2/27/17.
+ */
+public class GraphImpl implements Graph {
+
+    private static final ObjectMapper mapper = ObjectMapperHelper.get();
+    private static final int MAX_ATTEMPTS = 60;
+    private static final int SYNC_SLEEP_INTERVAL = 5000;
+    private final XrayClient xray;
+
+    public GraphImpl(XrayClient xray) {
+        this.xray = xray;
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        FilterProvider filters = new SimpleFilterProvider().setFailOnUnknownId(false)
+                .addFilter("xray-graph-filter", SimpleBeanPropertyFilter.filterOutAllExcept("component_id", "nodes"));
+        mapper.setFilterProvider(filters);
+    }
+
+    @Override
+    public GraphResponse graph(DependencyTree dependencies, String projectName) throws IOException, InterruptedException {
+        if (projectName == null || projectName.isEmpty()) {
+            return graph(dependencies);
+        }
+        if (dependencies == null) {
+            return new GraphResponseImpl();
+        }
+        return this.post("?project=" + projectName, dependencies);
+    }
+
+    @Override
+    public GraphResponse graph(DependencyTree dependencies) throws IOException, InterruptedException {
+        if (dependencies == null) {
+            return new GraphResponseImpl();
+        }
+        return this.post("", dependencies);
+    }
+
+    private GraphResponse post(String params, Object body) throws IOException, InterruptedException {
+        HttpEntity entity = null;
+        // First, request a scan from Xray.
+        try (CloseableHttpResponse response = xray.post("scan/graph" + params, body, mapper)) {
+            entity = response.getEntity();
+            GraphResponse requestResponse = mapper.readValue(response.getEntity().getContent(), GraphResponseImpl.class);
+            String scanId = requestResponse.getScanId();
+            // If no context was provided (project name), we would like to receive all known vulnerabilities.
+            String includeVulnerabilities = !params.isEmpty() ? "" : "&include_vulnerabilities=true";
+            // Xray will respond with 201 until the completion of the scan, then 200 will be returned.
```

Comment:
...scan, then 200 will be returned. --> ...scan. Once completed, 200 will be returned..